build/AzurePipelinesTemplates/WindowsAppSDK-RunTestsInPipeline-Job.yml

```diff
@@ -0,0 +1,120 @@
+# This runs the integration tests directly in the pipeline on the images hosted by the 'WinAppSDK-Test-Pool'.
+# This pool is a 1ES Hosted Pool with custom images, which lets us run tests on builds beyond what is
+# supported by Helix or the Azure DevOps Microsoft hosted agents. In case of arm64, we use the 
+# 'WinAppSDK-Test-Pool-Arm64' pool instead, and use parameter isArm64Platfrom to drive pool selection.
+
+parameters:
+- name: jobName
+  type: string
+  default: PipelineTests
+- name: isArm64Platfrom
+  type: boolean
+  default: false
+- name: dependsOn
+  type: object
+  default:
+  - ''
+  
+jobs:
+- job: ${{ parameters.jobName }}
+  dependsOn: ${{ parameters.dependsOn }}
+  strategy:
+    # Debug/Release buildConfiguration below refers to the build of the tests/apps. The OS image bits are always Release build.
+    matrix:
+      # The test matrix for x64 is currently larger; this is where we exercise most variations.
+      ${{ if not( parameters.isArm64Platfrom ) }}:
+        RS5_x86chk:
```

Comment:
We use chk builds? I thought those went the way of the dodo.

Reply:
This is the same list as machines from Aggregator.

Are chk builds valid? @alexlamtest / @kythant

Reply:
> This is the same list as machines from Aggregator.

Can we use a shared list across repos?

Should we?

If we can/should we add eng\TestEnvironments.xml with the various definitions, let Maestro flow that across the repos and make ADO use that list?

Reply:
The 'chk' refers to the **test binaries** and not the OS? That's a very different matter - and confusingAH (Windows builds come in fre + chk flavors, WinAppSDK binaries come in release + debug flavors)

I like the idea of running some debug test binaries, but not at the cost of not running release bits on that SKU. We ship release bits. We should be testing release bits on all supported platforms. If we ALSO want to run debug binaries on some SKUs I could agree with that. But if I'm reading this right we're running debug bits on RS5 and _not_ testing our release bits on RS5. Not cool.

Reply:
Renamed to FRE